portfolio/src/main/webapp/comments.js

```diff
@@ -16,28 +18,37 @@ async function fetchComments() {
       buildCommentsTree(commentsMap, comment, 0);
     }
   }
-  if (commentsJson.length) {
-    // if there are some comments, remove label 'No comments yet'
-    addDeleteButton();
-    document.getElementById('no-comments').remove();
+  if (!commentsJson.length) {
+    // if there are no comments, add tag 'no comments'
+    createNoCommentsTag();
   }
+  // if client is authenticated, add new comment form
+  if (authState.authenticated) {
+    createNewCommentForm();
+    if (commentsJson.length) {
+      // if there are some comments, add button to delete all of them
+      // only authenticated user can delete comments (maybe will change later to administrator only feature)
+      addDeleteButton();
```

Comment:
If you are implementing delete based on auth, then isn't it better to provide functionality such that you can delete a specific comment and not all of your comments?

Reply:
it is implemented in the further PR